src/http.rs

```diff
@@ -31,11 +34,25 @@ pub fn get_client() -> Client<Connector, hyper::Body> {
 pub fn fetch_sync_string(module_name: &str) -> DenoResult<String> {
   let url = module_name.parse::<Uri>().unwrap();
   let client = get_client();
-  let future = client
+  let fetch_future = client
     .get(url)
-    .and_then(|response| response.into_body().concat2());
-  let body = tokio_util::block_on(future)?;
-  Ok(String::from_utf8(body.to_vec()).unwrap())
+    .map_err(|err| DenoError::from(err))
+    .and_then(|response| {
+      if !response.status().is_success() {
+        return Either::A(futures::future::err(errors::new(
+          errors::ErrorKind::NotFound,
+          "module not found".to_string(),
+        )));
+      }
+      Either::B(
+        response
+          .into_body()
+          .concat2()
+          .map(|body| String::from_utf8(body.to_vec()).unwrap())
+          .map_err(|err| DenoError::from(err)),
+      )
```

Comment:
I tried to do this more simply myself but couldn't. I think this is correct. Rust is annoyingly painful at times.